inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/manager/MessageReadinessManager.kt

```diff
@@ -125,11 +125,12 @@ internal interface MessageReadinessManager {
                 )
 
         /**
-         * This method checks if the message displayed less than its max impressions,
-         * or message has been opted out.
+         * This method checks if the message has infinite impressions, or has been displayed less
+         * than its max impressions, or has been opted out.
          */
         private fun shouldDisplayMessage(message: Message): Boolean =
-            (LocalDisplayedMessageRepository.instance().numberOfTimesDisplayed(message)
+            (message.infiniteImpressions() ||
```

Comment:
If I'm not mistaken, the operator precedence here is `<`, then `&&` and then `||`. That would mean when campaigns have `infiniteImpressions()` they won't be checked against `LocalOptedOutMessageRepository`

Reply:
@imaginaris, the first check before && is into braces  (it doesn't appear in the PR but you can see it in the whole code block), so the LocalOptedOutMessageRepository.instance().hasMessage() is always checked.

Reply:
aaah, so the closing `)` is not at the end. I can see it now! No issue then